.github/workflows/_test_containerized.yml

```diff
@@ -92,34 +62,17 @@ jobs:
             -v "${HOME}/go/pkg/mod:/go/pkg/mod"                                 \
             -v "$PWD:$PWD"                                                      \
             -w "$PWD"                                                           \
-            -eCGO_ENABLED="${{ matrix.cgo-enabled }}"                           \
-            -eDD_APPSEC_WAF_LOG_FILTER="@ waf[.]cpp:"                           \
             -eDD_APPSEC_WAF_LOG_LEVEL="${{ matrix.waf-log-level }}"             \
-            -eDD_APPSEC_WAF_TIMEOUT="${{ matrix.waf-timeout || '5s' }}"         \
-            -eGODEBUG="${{ matrix.go-debug }}"                                  \
-            -eGOEXPERIMENT="${{ matrix.go-experiment }}"                        \
             -eGOMODCACHE="/go/pkg/mod"                                          \
             "${{ format(matrix.image, matrix.go-version) }}"
-      - name: Install alpine requirements
-        if: endsWith(matrix.image, '-alpine') && matrix.cgo-enabled == '1'
-        run: |-
-          docker exec -i gha-${{ github.run_id }}                               \
-            apk add gcc musl-dev libc6-compat
```

Comment:
I suggest you keep this here to do the same as the following al2 setup, and remove the setup out of ci.sh

Also, I find what I did here https://github.com/DataDog/dd-trace-go/pull/2541/files#diff-dec1b363c4670835b0b6e96b3cebe6711e688d0dd14701a2cffaf6e6d1f60480R226-R236 a bit simpler: everything in the same place instead of behind spread around multiple steps. It also simplifies the container creation and destruction.

Reply:
I can't put it back here because these packages needs to be installed when running with CGO and also specifically need to NOT be installed when running without CGO. Hence why, in my script, I run non-CGO stuff, then I install these packages, and finally I run the CGO stuff.

About what you did here ([DataDog/dd-trace-go#2541 (files)](https://github.com/DataDog/dd-trace-go/pull/2541/files#diff-dec1b363c4670835b0b6e96b3cebe6711e688d0dd14701a2cffaf6e6d1f60480R226-R236)) I feel that my script is a little too big to be directly embedded in yaml. Plus I would have to duplicate it since the same script is ran in the 2 different workflows (bare-metal & containerized)

Reply:
Sounds good.
So you actually mix "smoke tests" with these.